src/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -151,6 +145,20 @@ private function setDomDocument(\DOMDocument $domDocument): void
         $this->xPath = new \DOMXPath($this->domDocument);
     }
 
+    /**
+     * @return \DOMXPath
+     *
+     * @throws \BadMethodCallException
+     */
+    protected function getXPath(): \DOMXPath
+    {
+        if (!$this->xPath instanceof \DOMXPath) {
+            throw new \BadMethodCallException('$this->xPath has not been initialized yet.', 1617819086);
```

Comment:
Could we use `self::class` in the message (like for `getDomDocument`) rather than `$this` which might be a bit vague?

Reply:
Yes, we can! Changed and repushed.